### PR TITLE
Rewrite `create.js` in TypeScript

### PR DESCRIPTION
**Pull request by michaeldera**
#### After npm run build
I have one issue I encountered I am not really sure how to solve on line 67 and 68 of [create.ts](source/create.ts). It is generating the following errors:

- Element implicitly has 'any' type because type '{ (url: URL, options: Options): any; create: (defaults: any ) => ...; has no index signature, and 
- Element implicitly has 'any' type because type '( url: URL, options: Options) => any' has no index signature.

I am not sure how to approach the issue... however if, instead of using the `for (const method of aliases){` on line 66 and I assign each alias individually, i.e `got.put = (url: Url ....` the problem goes away but I presume there is a better way

### CODE REVIEW

**Comment by sindresorhus** [comment]
Can you do the first commit as just renaming `.js` to `.ts` using `git mv`? That way GitHub can show an actual diff between the files. Right now it's impossible to review the differences.

File: source/create.ts [hunk 1]
```diff
@@ -0,0 +1,82 @@
+'use strict';
+import {URL} from 'url';
```
Comment (sindresorhus) [inline source/create.ts]: Don't import `URL`. We use the `URL` global.

File: source/create.ts [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+import errors from './errors';
+import asStream from './as-stream';
+import asPromise from './as-promise';
+import { normalize, preNormalize } from './normalize-arguments';
+import merge, { mergeOptions, mergeInstances} from './merge';
+import deepFreeze from './utils/deep-freeze';
+import { InterfaceWithDefaults, Method, Options, NextFunction } from './utils/types';
+
+const getPromiseOrStream = ( options : any ) => options.stream ? asStream(options) : asPromise(options);
```
Comment (sindresorhus) [inline source/create.ts]: Don't use `any`. Type it properly.

File: source/create.ts [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+	defaults = merge({}, defaults);
+	preNormalize(defaults.options);
+
+	if (!defaults.handler) {
+		// This can't be getPromiseOrStream, because when merging
+		// the chain would stop at this point and no further handlers would be called.
+		defaults.handler = (options: any, next: NextFunction) => next(options);
+	}
+
+	function got(url: URL, options: Options) {
```
Comment (sindresorhus) [inline source/create.ts]: Got also supports a `string` in the `url` parameter.

**Comment by michaeldera** [comment]
Alright. Let me redo the PR.

**Comment by sindresorhus** [comment]
There's no need to close or open a new PR. You can just push additional commits to this PR ;)

File: source/create.ts (hunk 1 above)
Comment (szmarczak) [reply source/create.ts]: @sindresorhus Does TypeScript resolve it automatically if using Node 8?

**Comment by michaeldera** [comment]
> 
> 
> There's no need to close or open a new PR. You can just push additional commits to this PR ;)

Please forgive my ignorance... the first commit, where I am supposed to do `git mv` so it is possible to see the diff, that is the part I cannot get around adding that commit... tried rebasing but not quite getting that bit right

**Comment by szmarczak** [comment]
@michaeldera Ping. Are you finishing this PR? Otherwise, I could work on it :)

**Comment by michaeldera** [comment]
> @michaeldera Ping. Are you finishing this PR? Otherwise, I could work on it :)

If it is not too far off I can work on it over the weekend but you could work on it if you would like.

**Comment by szmarczak** [comment]
> If it is not too far off I can work on it over the weekend but you could work on it if you would like.

Sure! I'm glad I can help :) At the moment I'm focused on my other Got PRs - gotta fix them.

**Comment by michaeldera** [comment]
Great! That works! If I am able to do it before I certainly will :)

File: source/create.ts (hunk 1 above)
Comment (sindresorhus) [reply source/create.ts]: Never mind. I got confused with Ky.

**Comment by sindresorhus** [comment]
> Please forgive my ignorance... the first commit, where I am supposed to do `git mv` so it is possible to see the diff, that is the part I cannot get around adding that commit... tried rebasing but not quite getting that bit right

Copy-paste your changes somewhere as a backup, create a new branch from `master`, rename the `.js` file to `.ts`, commit this change, then copy-paste back your changes to the file, now commit these changes as a new commit, then force push to this remote PR branch.

**Comment by michaeldera** [comment]
Doing this over now

**Comment by sindresorhus** [comment]
No need to create a new PR. Force push is your friend.

**Comment by sindresorhus** [comment]
@michaeldera Still interested in finishing this?

**Comment by michaeldera** [comment]
I am sorry I will not be able to right now.

**Pull request closed without merging**